query/README.md

```diff
@@ -21,35 +21,183 @@ KeenQueryClient queryClient = new TestKeenQueryClientBuilder(queryProject)
 	.build();
 ```
 ### Using the KeenQueryClient to send Queries
-Users can use the KeenQueryClient to send queries as follows:
+The most simple way that users can use the KeenQueryClient to send queries is as follows. These methods take only the required query parameters as input, and the user receives a very specific Integer or Double response type. Please note that we strongly encourage users to pass in the Timeframe parameter, but it can be null.
 ```java
-Integer result = queryClient.count("<event_collection>");
+Integer count = queryClient.count("<event_collection>", new Timeframe("this_year"));
+Integer countUnique = queryClient.countUnique("<event_collection>", "<target_property>", new Timeframe("this_year"));
+Double minimum = queryClient.minimum("<event_collection>", "<target_property>", new Timeframe("this_year"));
+Double maximum = queryClient.maximum("<event_collection>", "<target_property>", new Timeframe("this_year"));
+Double average = queryClient.average("<event_collection>", "<target_property>", new Timeframe("this_year"));
+Double median = queryClient.median("<event_collection>", "<target_property>", new Timeframe("this_year"));
+Double percentile = queryClient.percentile("<event_collection>", "<target_property>", new Timeframe("this_year"));
+Double sum = queryClient.sum("<event_collection>", "<target_property>", new Timeframe("this_year"));
 ```
-Alternatively, users can use optional parameters via the object KeenQueryParams. However, because the result may be different depending on the optional parameters, the return value is an Object. The user is expected to verify the expected return type for the query, given the parameters entered.
+The exceptions are Select Unique, Extraction, Funnel, and Multi-Analysis queries which are more a little more complicated and are mentioned below (TODO: Mention below!!!).
+
+### Advanced
+Alternatively, users can use optional parameters to send queries. However, because the result may be different depending on the optional parameters, the return value is a QueryResult. The user is expected to verify the expected return type for the query, given the parameters entered.
 ```java
-KeenQueryParams queryParams = new QueryParamBuilder()
+Query query = new QueryBuilder(QueryType.COUNT_RESOURCE)
+        .withEventCollection("<event_collection>")
+        .withTimeframe(new Timeframe("<start>", "<end>"))
+        .build();
+QueryResult result = queryClient.execute(query, new Timeframe("this_month"));
```

Comment:
Oops. My fault with copy/pasting from the old test cases. I will be removing Timeframe entirely from the Query class, since we will force the user to specify the Timeframe when executing the Query. I'll fix this in the code and the README.